styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -174,12 +190,17 @@ private void initialize(WorkflowInstance workflowInstance) {
       if (e.isAlreadyExists()) {
         throw new AlreadyInitializedException("Workflow instance is already triggered: " + workflowInstance);
       } else if (e.isConflict()) {
-        LOG.debug("Transactional conflict, abort triggering Workflow instance: " + workflowInstance);
+        log.debug("Transaction conflict when triggering workflow instance. Aborted: {}",
+            workflowInstance);
         throw new RuntimeException(e);
       } else {
+        log.debug("Transaction failure when triggering workflow instance: {}: {}",
+            workflowInstance, e.getMessage(), e);
         throw new RuntimeException(e);
       }
-    } catch (IOException e) {
+    } catch (Exception e) {
+      log.debug("Failure when triggering workflow instance: {}: {}", workflowInstance, e.getMessage(), e);
+      Throwables.propagateIfPossible(e, RuntimeException.class);
```

Comment:
This method is deprecated.

Reply:
`Throwables.propagate()` is deprecated, but `Throwables.propagateIfPossible()` is not. However, the newer `Throwables.throwIfUnchecked()` might be preferrable to use.